crates/shim-sgx/src/handler/mod.rs

```diff
@@ -78,18 +78,20 @@ pub struct Handler<'a> {
 
 impl<'a> Handler<'a> {
     /// Acknowledge pages committed by the host with ENCLS[EAUG].
-    fn accept_mmap(&mut self, region: Region) {
+    fn accept_mmap(&mut self, region: Region, prot: c_int) {
         assert!(region.start < region.end);
         let length = region.end - region.start;
-        let secinfo = Class::Regular.info(Flags::READ | Flags::WRITE | Flags::EXECUTE);
+        let flags = Flags::from_bits_truncate((prot | PROT_READ) as u8);
         let zero = PageAddr::from_start_address(VirtAddr::new(ZERO.as_ptr() as u64)).unwrap();
+        // TODO: Remove ` | PROT_READ` when possible. It's only done  because
+        // some of the ioctl's in the v4 of the SGX2 series check for it.
         for i in 0..length.items() {
             let virt_addr = VirtAddr::new((region.start.raw() + i * Page::SIZE) as u64);
             let page_addr = PageAddr::from_start_address(virt_addr).unwrap();
-            // FIXME: handle with `.unwrap()` once `SecInfo` implements `Debug` trait.
-            if secinfo.accept_copy(page_addr, zero).is_err() {
-                panic!();
-            }
+            Class::Regular
+                .info(flags)
+                .accept_copy(page_addr, zero)
+                .unwrap();
```

Comment:
We should try very hard to avoid `.unwrap()` in the shim. What is the actual failure case here?